apps/studio/src/pages/sign-in/singpass/index.tsx

```diff
@@ -1,30 +1,58 @@
+import { useEffect, useState } from "react"
+import NextLink from "next/link"
 import { useRouter } from "next/router"
-import { Flex, Grid, GridItem, Text, useDisclosure } from "@chakra-ui/react"
-import { Button, RestrictedGovtMasthead } from "@opengovsg/design-system-react"
+import {
+  Box,
+  Flex,
+  Grid,
+  GridItem,
+  Link,
+  Text,
+  useDisclosure,
+} from "@chakra-ui/react"
+import {
+  Button,
+  Infobox,
+  RestrictedGovtMasthead,
+} from "@opengovsg/design-system-react"
 import { BiChevronLeft } from "react-icons/bi"
 
 import type { NextPageWithLayout } from "~/lib/types"
 import { PublicPageWrapper } from "~/components/AuthWrappers"
 import { FullscreenSpinner } from "~/components/FullscreenSpinner"
 import { IsomerLogo } from "~/components/Svg"
+import { ISOMER_SUPPORT_LINK } from "~/constants/misc"
 import { SingpassLoginButton } from "~/features/sign-in/components"
 import { UnableToUseSignpassModal } from "~/features/sign-in/components/UnableToUseSignpassModal"
 import { SIGN_IN } from "~/lib/routes"
 import { trpc } from "~/utils/trpc"
 
 const SingpassSignInPage: NextPageWithLayout = () => {
+  const [isSingpassError, setIsSingpassError] = useState<boolean>()
   const {
     isOpen: isUnableToUseSingpassModalOpen,
     onOpen: openUnableToUseSingpassModal,
     onClose: closeUnableToUseSingpassModal,
   } = useDisclosure()
   const router = useRouter()
-  const { data, isLoading } = trpc.auth.singpass.getName.useQuery()
+  const { data, isLoading, isError } = trpc.auth.singpass.getName.useQuery()
 
   const handleBackToLogin = async () => {
     await router.push(SIGN_IN)
   }
 
+  useEffect(() => {
+    if (isError) {
+      void router.push(`${SIGN_IN}?error=Unable to retrieve user data`)
```

Comment:
nit - there's no issue with this since we are the one passing in the error message, but might wanna wrap the error message in `encodeURIComponent` as a best practise

Reply:
Updated in a116f4da5ed9d10d8d78ae1f8bab668ac4f67b04.